### PR TITLE
feat(admin): set login for existing service account

### PR DESCRIPTION
**Pull request by HejdaJakub**
* If user creates service account without login, now he/she can set the login for some namespace
later. This functionality is accessible on the service member overview page.

### CODE REVIEW

File: apps/admin-gui/src/app/shared/components/create-service-member-dialog/create-service-member-dialog.component.ts [hunk 1]
```diff
@@ -152,25 +130,8 @@ export class CreateServiceMemberDialogComponent implements OnInit, AfterViewInit
...
-          this.usersManagerService
-            .isLoginAvailable(namespace, control.value as string)
-            .pipe(map((res) => (res ? null : { loginExists: true })))
-        )
-      );
-    };
-  }
-
   onCreate(): void {
+    this.cd.detach();
```
Comment (bodnara) [inline apps/admin-gui/src/app/shared/components/create-service-member-dialog/create-service-member-dialog.component.ts:134]: Not sure why we detach here. When detached the component will not update with changes e.g. loading indicator is not shown because of this.

File: apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
+<div class="vo-theme">
+  <h1 class="mat-dialog-title">{{'DIALOGS.SET_LOGIN.TITLE' | translate}}</h1>
+  <div *ngIf="!processing" class="dialog-container" mat-dialog-content>
+    <app-perun-web-apps-login-for-namespace
+      [formGroup]="formGroup"
+      (namespaceChanged)="selectedNamespace = $event">
+    </app-perun-web-apps-login-for-namespace>
+  </div>
+  <mat-spinner *ngIf="processing" class="ml-auto mr-auto"></mat-spinner>
+  <div *ngIf="!processing" mat-dialog-actions>
```
Comment (bodnara) [inline apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.html:10]: I believe we don't hide the dialog actions, just disable the "action" button.

File: apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.ts [hunk 3]
```diff
@@ -0,0 +1,70 @@
+import { Component, Inject, OnInit } from '@angular/core';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { User, UsersManagerService } from '@perun-web-apps/perun/openapi';
```
Comment (bodnara) [inline apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.ts:4]: `User` is never used.

File: apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.ts [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+})
+export class SetLoginDialogComponent implements OnInit {
+  formGroup: FormGroup;
+  processing = false;
+  selectedNamespace: string;
+
+  constructor(
+    private dialogRef: MatDialogRef<SetLoginDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) private data: SetLoginDialogComponentData,
+    private _formBuilder: FormBuilder,
```
Comment (bodnara) [inline apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.ts:25]: We don't use this underscore notation.

File: apps/admin-gui/src/app/vos/pages/member-detail-page/member-overview/member-overview.component.ts [hunk 5]
```diff
@@ -38,6 +40,8 @@ export class MemberOverviewComponent implements OnInit {
   vo: Vo;
   loading = false;
   pwdResetAuth: boolean;
+  // TODO: authorization
+  setLoginAuth = true;
```
Comment (bodnara) [inline apps/admin-gui/src/app/vos/pages/member-detail-page/member-overview/member-overview.component.ts:44]: The comment can be removed I believe, also maybe the default value should be `false`.

**Review by bodnara — CHANGES_REQUESTED** [review]

File: apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.html (hunk 2 above)
Comment (HejdaJakub) [reply apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.html]: Fixed.
(NOTE: I used the `create-service-member-dialog` as a template, so we sometimes use it, but I agree that just disabling it is a better solution.)

File: apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.ts (hunk 4 above)
Comment (HejdaJakub) [reply apps/admin-gui/src/app/shared/components/set-login-dialog/set-login-dialog.component.ts]: Fixed.
(NOTE: Again I used the `create-service-member-dialog` as a template.)

**Review by xkureck — CHANGES_REQUESTED** [review]
Move the section to service detail section and fix creating login for mu

File: apps/admin-gui/src/app/shared/components/login-password-form-with-generate-option/login-password-form-with-generate-option.component.ts [hunk 6]
```diff
@@ -0,0 +1,75 @@
+import { Component, EventEmitter, Input, Output } from '@angular/core';
+import { FormGroup, Validators } from '@angular/forms';
+import { enableFormControl } from '@perun-web-apps/perun/utils';
+import { loginAsyncValidator } from '@perun-web-apps/perun/namespace-password-form';
+import { MembersManagerService, UsersManagerService } from '@perun-web-apps/perun/openapi';
+import { ApiRequestConfigurationService } from '@perun-web-apps/perun/services';
+
+@Component({
+  selector: 'app-perun-web-apps-login-password-form-with-generate-option',
```
Comment (bodnara) [inline apps/admin-gui/src/app/shared/components/login-password-form-with-generate-option/login-password-form-with-generate-option.component.ts:9]:
```suggestion
  selector: 'app-login-password-form-with-generate-option',
```

**Review by bodnara — CHANGES_REQUESTED** [review]
Overall well done, only couple of selector inconsistencies.

**Review by bodnara — APPROVED** [review]

**Review by xkureck — CHANGES_REQUESTED** [review]
image
just one little detail - the tooltip is a little bit misleading - maybe we don't need the tooltip on this selected option or change the text of the tooltip
after fixing this merge it please
